### PR TITLE
Add docs for TLS Server Certificate rotation for K8S

### PR DESCRIPTION
**Pull request by thisisnotashwin**

### CODE REVIEW

File: website/data/docs-navigation.js
Comment (ishustava) [inline website/data/docs-navigation.js:198]: I'm thinking that maybe we should call it just "certificate rotation" to leave space for other cert rotation docs, e.g. CAs. wdyt?

File: website/content/docs/k8s/operations/server-certificate-rotation.mdx
Comment (ishustava) [inline website/content/docs/k8s/operations/server-certificate-rotation.mdx:11]: I'm thinking that tls-init might be an internal implementation that folks are not familiar with
```suggestion
are issued every time the Helm chart is upgraded. These certificates are signed by the same CA and will
```

File: website/content/docs/k8s/operations/server-certificate-rotation.mdx
Comment (ishustava) [inline website/content/docs/k8s/operations/server-certificate-rotation.mdx:14]:
```suggestion
Consul servers read the certificates from Kubernetes secrets during start-up and keep them in memory. In order to ensure the
```

File: website/content/docs/k8s/operations/server-certificate-rotation.mdx
Comment (ishustava) [inline website/content/docs/k8s/operations/server-certificate-rotation.mdx:15]:
```suggestion
servers use the newer certificate, the server pods need to be [restarted safely](/docs/k8s/operations/upgrade#upgrading-consul-servers).
```

File: website/content/docs/k8s/operations/server-certificate-rotation.mdx
Comment (lkysow) [inline website/content/docs/k8s/operations/server-certificate-rotation.mdx:11]: Should we say on every helm upgrade rather than every time the TLS-init job runs? Or maybe "every time the TLS-init job runs, which happens on every helm upgade"?

I think most folks won't know what that job is.

File: website/content/docs/k8s/operations/server-certificate-rotation.mdx
Comment (lkysow) [inline website/content/docs/k8s/operations/server-certificate-rotation.mdx:14]:
```suggestion
Consul servers read the certificates during start-up and keep them in-memory. In order to ensure the
```

File: website/content/docs/k8s/operations/server-certificate-rotation.mdx
Comment (ishustava) [reply website/content/docs/k8s/operations/server-certificate-rotation.mdx:15]: I think we should say that in some cases a helm upgrade will not cause an upgrade of the consul servers, and in that case the servers will not pick up the new certificates.

File: website/content/docs/k8s/operations/server-certificate-rotation.mdx
Comment (lkysow) [inline website/content/docs/k8s/operations/server-certificate-rotation.mdx:17]:
```suggestion
To explicitly perform certificate rotation, perform the following steps:
```

**Review by lkysow — APPROVED** [review]
Should we also mention that the CA cert gets renewed?

File: website/content/docs/k8s/operations/server-certificate-rotation.mdx
Comment (ishustava) [inline website/content/docs/k8s/operations/server-certificate-rotation.mdx:17]:
```suggestion
To explicitly perform server certificate rotation, follow these steps:
```

**Review by ishustava — COMMENTED** [review]
Thanks for adding these docs, Ashwin! I left some comments, but my main question is how should we treat upgrades where servers are going to be restarted anyways vs the ones where servers are not upgraded. If servers will be upgraded anyway, then users should follow the normal upgrade procedure. If not, then maybe suggest running upgrade with `--dry-run` and then follow the steps you outlined.

File: website/content/docs/k8s/operations/server-certificate-rotation.mdx
Comment (ishustava) [reply website/content/docs/k8s/operations/server-certificate-rotation.mdx]: jinx! I've added the same comment 😄

File: website/data/docs-navigation.js
Comment (thisisnotashwin) [reply website/data/docs-navigation.js:198]: that makes sense. I wanted to be explicit for now hoping that in the future, when we add more details to this page regarding cert rotation for other certificates (CA certs), we would update it, but now seems as good as later

File: website/content/docs/k8s/operations/certificate-rotation.mdx
Comment (ishustava) [inline website/content/docs/k8s/operations/certificate-rotation.mdx:5]: Let's update these titles too to make it less specific to servers

**Review by ishustava — APPROVED** [review]

**Comment by thisisnotashwin** [comment]
Will merge this after #9625 gets merged.

**Pull request merged**
